topics/go/testing/benchmarks/caching/caching.go

```diff
@@ -27,26 +27,17 @@ type data struct {
 var list *data
 
 func init() {
-	var last *data
-
 	// Create a link list with the same number of elements.
 	for row := 0; row < rows; row++ {
 		for col := 0; col < cols; col++ {
 
-			// Create a new node and link it in.
-			var d data
-			if list == nil {
-				list = &d
-			}
-			if last != nil {
-				last.p = &d
-			}
-			last = &d
+			// Create a new node and link it backwards.
+			list = &data{p: list}
```

Comment:
Yea, since list is nil at the beginning this should work. I'm guessing you have tested this?